### PR TITLE
[clang-scan-deps] Fix check for empty `Compilation`

### LINKED ISSUE #64144: clang-scan-deps crashes when input file does not exist
Simple repro:
```
clang-scan-deps --format=p1689 -- `which clang++` --std=c++20 this-file-does-not-exist.cpp
```

Result:
> ```
> error: no such file or directory: 'this-file-does-not-exist.cpp'
> error: no input files
> PLEASE submit a bug report to https://github.com/Homebrew/homebrew-core/issues and include the crash backtrace.
> Stack dump:
> 0.	Program arguments: clang-scan-deps --format=p1689 -- /opt/homebrew/opt/llvm/bin/clang++ --std=c++20 this-file-does-not-exist.cpp
> Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
> 0  libLLVM.dylib            0x0000000114052494 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
> 1  libLLVM.dylib            0x0000000117e9d550 SignalHandler(int) + 304
> 2  libsystem_platform.dylib 0x000000019880d4c4 _sigtramp + 56
> 3  clang-scan-deps          0x0000000102c4aae4 main + 5448
> 4  dyld                     0x0000000102df1088 start + 516
> [1]    25079 segmentation fault  clang-scan-deps --format=p1689 -- /opt/homebrew/opt/llvm/bin/clang++ 
> ```

Version info:
> ```
> $ clang-scan-deps --version
> Homebrew LLVM version 16.0.6
>   Optimized build.
> ```

I can reproduce this with non-Homebrew builds, so I report here instead of to Homebrew.

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes https://github.com/llvm/llvm-project/issues/64144

Instead of checking for `nullptr`  we need to ensure that `JobList` is not empty to proceed

### CODE REVIEW

**Comment by Yaraslaut** [comment]
ping

File: clang/tools/clang-scan-deps/ClangScanDeps.cpp [hunk 1]
```diff
@@ -728,7 +728,7 @@ getCompilationDataBase(int argc, char **argv, std::string &ErrorMessage) {
                            *Diags);
   std::unique_ptr<driver::Compilation> C(
       TheDriver.BuildCompilation(CommandLine));
-  if (!C)
+  if (C->getJobs().empty())
```
Comment (ChuanqiXu9) [inline clang/tools/clang-scan-deps/ClangScanDeps.cpp:731]:
```suggestion
  if (!C || C->getJobs().empty())
```

**Review by ChuanqiXu9 — COMMENTED** [review]
It will be better to have a test for this.

File: clang/tools/clang-scan-deps/ClangScanDeps.cpp (hunk 1 above)
Comment (Yaraslaut) [reply clang/tools/clang-scan-deps/ClangScanDeps.cpp]: Corrected

**Review by jansvoboda11 — APPROVED** [review]
LGTM, thanks!

**Comment by Yaraslaut** [comment]
> It will be better to have a test for this.

I added lit test

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM then.

**Pull request merged**